src/common/image.cpp

```diff
@@ -247,11 +253,16 @@ wxObjectRefData* wxImage::CloneRefData(const wxObjectRefData* that) const
     unsigned size = unsigned(refData->m_width) * unsigned(refData->m_height);
     if (refData->m_alpha != NULL)
     {
-        refData_new->m_alpha = (unsigned char*)malloc(size);
-        memcpy(refData_new->m_alpha, refData->m_alpha, size);
+       // Note allocator might have changed from previous image to this image - careful.
+        refData_new->m_alpha = (unsigned char*)refData_new->m_alloc->Alloc(size);
+        if (refData_new->m_alpha) {
+           memcpy(refData_new->m_alpha, refData->m_alpha, size);
+        }
+        // else refData_new->m_alpha is NULL, because memory alloc failed. 
     }
     size *= 3;
-    refData_new->m_data = (unsigned char*)malloc(size);
+    refData_new->m_data = (unsigned char*)refData_new->m_alloc->Alloc(size);
+    // TODO m_data == NULL
```

Comment:
If _m_data == NULL_ is possible, it should be handled? Similarly for alpha below.

Reply:
I don't think wxWidgets code checks for memory allocation failure anywhere.
So I will leave it untouched.